File: 21560 - Sai Suraj/week 1/1598_Crawler Log Folder.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        curr_ops = 0
        for i in range(len(logs)):
            if logs[i] == "../":
                if curr_ops!=0:
                    curr_ops -= 1
                else:
                    continue
            elif logs[i]=='./':
                continue
            else:
                curr_ops+=1
        return curr_ops

File: 21558 - Nithin/week 1/1598-crawler-log-folder/1598-crawler-log-folder.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int no_op = logs.size();
        if(no_op == 0)
            return 0;
        int level = 0;
        if(logs[0] == "../" || logs[0] == "./")
            level = 0;
        else
            level = 1;
        for(int i = 1; i < no_op; i++){
            if(logs[i] == "../"){
                if(level != 0)
                    level -= 1;
                else
                    continue;
            }
            else if(logs[i] == "./")
                continue;
            else
                level += 1;
        }
        return level;
    }
};